[CS2113-T11-4] Inventra

tayjerom opened the pull request:
Inventra is a command-driven logistics and inventory management system design for small to medium-sized businesses including but not limited to: warehouses, online shops, retail businesses. Optimized for usage via Command Line Interface (CLI), Inventra aims to provide precise inventory management, real-time tracking and automation of tasks.

tayjerom committed:
Fix bugs pertaining to duplicate headers fields, invalid field type and -hu

tayjerom committed:
Merge branch 'master' into add-h-check

tayjerom committed:
Update CSV file path

tayjerom committed:
Merge pull request #24 from tayjerom/add-h-check

Fix bugs pertaining to duplicated fields and invalid file types on "add -h" command

tayjerom committed:
Implement update to csv after deleting records + gradle fix

tayjerom committed:
Merge pull request #25 from tayjerom/update-csv-after-deletion

Implement update to csv after deleting records + Gradle fix

danusan-s committed:
Added number related exceptions

danusan-s committed:
Added missing arguements exception

danusan-s committed:
Made Delete command use Exceptions

danusan-s committed:
Merge pull request #26 from danusan-s/refactoring-exceptions

Refactoring exceptions

danusan-s committed:
Added Delete Junit test

danusan-s committed:
Modified test method name

danusan-s committed:
Merge pull request #27 from danusan-s/added-delete-test

Added delete test

tayjerom committed:
Debug CSV header field types upon reload program, via assertion and logging

tayjerom committed:
Merge remote-tracking branch 'upstream/master' into fix-bugs-csv-header

Lambom4n committed:
Modify view command to use Inventra Exception

tayjerom committed:
Merge pull request #28 from tayjerom/fix-bugs-csv-header

Fix bugs pertaining to CSV Header

tayjerom committed:
Fix test-setup to create test CSV file and handle file pathing

DucPhong135 committed:
Merge pull request #29 from AY2425S1-CS2113-T11-4/branch-#29

Modify view command to use Inventra Exception

tayjerom committed:
Merge pull request #30 from tayjerom/fix-test-filepath

Fix test-setup to create test CSV file and handle file pathing

danusan-s committed:
Cleaned up view and delete

danusan-s committed:
Added a lot more exceptions

danusan-s committed:
Cleaned up ui class

danusan-s committed:
Updated junit tests

danusan-s committed:
Temporarily removed absolute path

danusan-s committed:
Merge pull request #31 from danusan-s/more-exceptions

More exceptions

danusan-s committed:
Added ability to delete entire table and all records

danusan-s committed:
Added ability to delete an entire column

danusan-s committed:
Fixed style for delete command

danusan-s committed:
Merge pull request #33 from danusan-s/delete-headers

Delete headers and table

Commit:
Merge remote-tracking branch 'Upstream_repo/master' into Delete-Headers

Lambom4n committed:
Add a new exception for too many argument being passed to a command and modify view command to use that exception

DucPhong135 committed:
Merge pull request #34 from DucPhong135/master

Add new exception for too many arguments passed to a command

danusan-s committed:
Modified the extra args exception

danusan-s committed:
Added help command and removed exit command

danusan-s committed:
Modified text-ui-test

danusan-s committed:
Merge pull request #35 from danusan-s/help-command

Help command

Commit:
Standardise expected output between cmd and IntelliJ

Commit:
Merge remote-tracking branch 'Upstream_repo/master' into merge-v0.2

Commit:
Test relative path

Commit:
Standardize intelliJ and cmd filepath

lucas-sc0 committed:
Merge pull request #36 from lucas-sc0/lucas-merge-0.2

Lucas merge 0.2

Commit:
Fix handling of metadata upon deletion

Commit:
Update csv properly after delete -a & delete -e

Commit:
Add new delete command flag variants into help command

lucas-sc0 committed:
Merge pull request #38 from lucas-sc0/master

Fix issue with delete -a and delete -e

Commit:
Comment out unecessary debugging info

lucas-sc0 committed:
Merge pull request #39 from lucas-sc0/master

Comment out unecessary debugging info

danusan-s committed:
Test coverage for entire delete command

danusan-s committed:
Fixed checkstyle

danusan-s committed:
Merge pull request #40 from danusan-s/delete-test-coverage

Test coverage for entire delete command

danusan-s committed:
Make file path simpler and jar compatible

danusan-s committed:
Modify text-ui-test script to clear up data folder upon completion

danusan-s committed:
Update delete test

danusan-s committed:
Reformatting to fix style

danusan-s committed:
Merge pull request #41 from danusan-s/file-path-fix

Got rid of complex file path system

danusan-s committed:
Added command specific help

danusan-s committed:
Merge pull request #42 from danusan-s/better-help-command

Added command specific help

danusan-s committed:
Added delete range feature

danusan-s committed:
Formatted code to pass checkstyle

danusan-s committed:
Merge pull request #43 from danusan-s/delete-range

Delete range

tayjerom committed:
Enable assertion in build.gradle, removed add -l command

tayjerom committed:
Update InventraTest: Replace add -l with view -a to fix test failures

tayjerom committed:
Merge pull request #45 from tayjerom/v2.0-start-v1

Enable assertion in build.gradle, removed add -l command

tayjerom committed:
Update AboutUs.md

Append personal GitHub information to team profiles.

Lambom4n committed:
add view command view -f

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

DucPhong135 committed:
Merge pull request #46 from DucPhong135/master

add view command by keyword view -f

Commit:
Add assertions

- Constructor Assertions
- Argument Length in execute()
- Field Count Matching in handleAddRecord()
- Null and Empty Checks in Critical Parts: Validates important values (e.g., fieldData, type, value) are not null or empty when being processed.

tayjerom committed:
Update DeveloperGuide.md

Populate general details of Developer Guide (DG) consisting of:
- Acknowledgement
- Product Scope: Target user profile + Value Proposition
- User Stories (updated)
- Non-Functional Requirements
- Glossary
- Instructions for manual testing (WIP)

Included enhancement documentation for [AddCommand]
- with Architecture, Sequence and Class Diagram (WIP)

tayjerom committed:
Update DeveloperGuide.md

Changes made to include (WIP) annotations.

Commit:
Repackage add command test

LuxLucky7 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

Commit:
Merge branch 'master' of https://github.com/LuxLucky7/tp

lucas-sc0 committed:
Merge pull request #52 from lucas-sc0/test-merge-0.2

Refactor add testing

Commit:
Improve add command testing

lucas-sc0 committed:
Merge pull request #53 from lucas-sc0/test-merge-0.2

Improve add command testing

Commit:
Add exception class

lucas-sc0 committed:
Merge pull request #54 from lucas-sc0/test-merge-0.2

Add exception class

tzerbin commented on src/main/java/seedu/command/AddCommand.java:
```diff
@@ -0,0 +1,226 @@
+        assert flag != null && !flag.isEmpty() : "Flag should not be null or empty";
+
+        switch (flag) {
+        case "-h":
```
Have these string cases be stored as a constant string; avoid magic strings / numbners.
This is apparent throughout the codebase.

tzerbin commented on src/main/java/seedu/command/AddCommand.java:
```diff
@@ -0,0 +1,226 @@
+        ui.showSuccessRecordAdded();
+    }
+
+    private String validateValue(String value, String type, String field) throws InventraException {
```
The method name validateValue did not hint to me that it will be returning a String.
Consider renaming the method / break down the method into smaller parts.

tzerbin commented on src/main/java/seedu/command/HelpCommand.java:
```diff
@@ -0,0 +1,95 @@
+            command += " " + parts[2];
+        }
+
+        switch (command) {
```
Consider breaking the cases down into just one case.
e.g. `case "add"` + `"add -h"` + ... can be simplified into just one case.
Also remember to use the `// Fallthrough` comment.

tzerbin commented on src/main/java/seedu/command/ViewCommand.java:
```diff
@@ -0,0 +1,112 @@
+        String keyword = sb.toString();
+        List<Map<String, String>> printedRecords = new ArrayList<>();
+        List<Integer> printedRecordsIndexes = new ArrayList<>();
+        for(String field : fields) {
```
This is a lot of nesting levels.
Consider simplifying the code.

tzerbin commented on src/main/java/seedu/storage/Csv.java:
```diff
@@ -0,0 +1,207 @@
+    }
+
+    // Load records from the CSV file into the Inventory
+    public void loadInventoryFromCsv(Inventory inventory) {
```
This is a very long method; consider breaking the method up into smaller pieces.

tzerbin reviewed (commented):
Relatively clean code.
However, the code base is also relatively small at this point in time.
Please remember to PR/push your work to your team's repo's master branch.

danusan-s committed:
Complete test coverage

danusan-s committed:
Added more help for new flags

danusan-s committed:
Cleanup to fit checkstyle

Change add command test method names and reformatted viewCommand and
DeleteCommandTest

danusan-s committed:
Merge pull request #55 from danusan-s/improve-delete

Improve delete test

danusan-s committed:
Update user guide to include delete command

danusan-s committed:
Added some info about delete in Developer Guide

danusan-s committed:
Merge pull request #56 from danusan-s/developer-guide

Developer guide

Oshen27 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* handleAddMultipleFields(): processes input fields, validate fields, and update the inventory accordingly.
+* handleAddRecord(): handles logic for adding records into the inventory. If successfully add records, data will be appended into the CSV file.
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](link-to-class-diagram)
```
Diagrams are not being displayed

Oshen27 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### "Add" Command Feature
+The "AddCommand" is responsible for adding fields and records to the inventory. 
+Concurrently, "add" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+This command supports the following options:
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+* '-hu': Updates fields in the inventory.
```
You may consider including an example of how the command looks like in this section

Oshen27 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
you may omit this

Oshen27 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](link-to-class-diagram)
+
+#### Sequence Diagram
+*Illustrates how "AddCommand" interacts with "Inventory" and "Csv" classes when adding a record:
+
+![Sequence Diagram](link-to-sequence-diagram)
```
You may consider adding an explanation for your diagrams, to help the reader understand the command better

Commit:
Update developer guide

Commit:
Add abstract command class

AndrewNguyen4 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* Java Coding Standard (Basic): [Official Documentation](https://se-education.org/guides/conventions/java/basic.html)
+* Code Quality (Guide): (https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html)
```
Hyperlink not working (not clickable in the DG)

AndrewNguyen4 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
 
 ## Design & implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```
This line is part of the template project and probably should be deleted when you work on your DG.

AndrewNguyen4 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
+![Architecture Diagram](link-to-architecture-diagram)
```
Images and diagrams in this DG are not displayed properly.

JuvenLimHY replied:
Diagram is not being displayed in the DG

AndrewNguyen4 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### "Add" Command Feature
```
Inconsistent formatting styles: "Add" (line 11),  '-h' (line 15), `command` (line 20). It is recommended that you use a single style for code snippets, most preferably `...`.

AndrewNguyen4 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+
+![Sequence Diagram](link-to-sequence-diagram)
+
+#### Why Implement "DeleteCommand" in this way?
+Separating implementation of "DeleteCommand", "Inventory", and "Csv" classes ensures:
+* "DeleteCommand" focus only on processing input and delegating task to other specific components.
+* "Csv" class is responsible for updating data into CSV file for storage; ensuring persistence in data control.
+* With this breakdown of implementation, create room for future scalability (e.g. adding more fields types).
+
+#### Alternative Considered
+Only when requirements permits, database implementation has been considered for better handling of fields types for inventory storage.
```
This section is highly similar with that of AddCommand, causing redundancy.
My recommendation: Remove the explanation at the end of AddCommand, and explain everything at the end of DeleteCommand.

ThienDuc3112 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+#### Sequence Diagram
+*Illustrates how "DeleteCommand" interacts with "Inventory" and "Csv" classes when adding a record:
+
+![Sequence Diagram](link-to-sequence-diagram)
```
None existent sequence diagram

ThienDuc3112 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* fields and records are validated before updating inventory data into the CSV file.
+
+"DeleteCommand" Methods:
+  ![Class Diagram](link-to-class-diagram)
```
None existent class diagram

JuvenLimHY replied:
Class Diagram is not shown

ThienDuc3112 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'DeleteCommand' can be visualized with the following architecture diagram:
+![Architecture Diagram](link-to-architecture-diagram)
```
None existent Architecture diagram

ThienDuc3112 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* '-a': Delete all records in the inventory. 
+* '-e': Delete all records and headers in the inventory.
+* '-h \<field_name>': Delete the mentioned field and its column from the inventory.
+* '-r \<start>-\<end>': Delete records from the start index to the end index (both inclusive and 1-based indexing).
```
Should have example below every command for clarity instead of in the manual testing part at the bottom of the document

ThienDuc3112 reviewed (commented):
it's alright

JuvenLimHY commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### "Add" Command Feature
+The "AddCommand" is responsible for adding fields and records to the inventory. 
+Concurrently, "add" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+This command supports the following options:
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+* '-hu': Updates fields in the inventory.
+
+#### Architecture Overview
+The 'AddCommand' is part of the `command` package, and interacts with the following key components:
+* **Inventory**: Contains the current state of the inventory, including fields and records.
+* **Csv**: Handles reading from and writing to the CSV file for data storage persistence.
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
+![Architecture Diagram](link-to-architecture-diagram)
+
```
Maybe add an use example to show how to execute the command

JuvenLimHY commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+* Java Coding Standard (Basic): [Official Documentation](https://se-education.org/guides/conventions/java/basic.html)
+* Code Quality (Guide): (https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html)
 
 ## Design & implementation
```
Can add a brief overview of the whole program to allow user view the whole sequence of the project

JuvenLimHY commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,152 @@
+
+#### Alternative Considered
+Only when requirements permits, database implementation has been considered for better handling of fields types for inventory storage.
 
```
Can add ur logging and exception to show how u catch errors

JuvenLimHY reviewed (commented):
I think there is some issue with the Diagram and a hyperlink. also can add more details.

Commit:
Update code to use abstract command class

Commit:
Remove arrowhead code

- Replaced with a single loop over records

Commit:
Improve output of view -f

- Standardize output to table format

Commit:
Improve add command testing

lucas-sc0 committed:
Merge pull request #57 from lucas-sc0/More-OOP

More oop

Commit:
Implement CommandParserTest

lucas-sc0 committed:
Merge pull request #59 from lucas-sc0/More-OOP

Implement CommandParserTest

Lambom4n committed:
add ViewCommandTest

DucPhong135 committed:
Merge pull request #60 from DucPhong135/master

add ViewCommandTest

tayjerom committed:
Update UserGuide to include guide for Add, View, Exit commands

Changes made to the User Guide includes:
- Refine Introduction, Quick Start, FAQ, Known Issues, and Command Summary
- Included details for Add, View, Exit command

tayjerom committed:
Update DeveloperGuide to include UML diagrams.

- Included section on 'ViewCommand' with its respective details and  UML diagram
- Included UML diagrams via generated PlantUML annotations.

tayjerom committed:
Update Developer Guide to include UML diagrams and View command

tayjerom committed:
Merge remote-tracking branch 'upstream/master' into v2.0-Week11

tayjerom committed:
Merge pull request #61 from tayjerom/v2.0-Week11

Update Developer Guide to include UML diagrams and View command

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
 # Developer Guide
```
[CRITICAL]
Diagrams are not shown on DG html page. Please fix this as soon as possible.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Main Components of the architecture
```
Perhaps include an architecture diagram here.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+* **Inventory**: The mode we are operating in.
+* **Storage**: Reads and writes data to a CSV file to maintain data persistence.
+
+### "Add" Command Feature
```
Do note that the DG's primary purpose is to inform on the implementation details.
However, the header seems to suggest to inform on what features have been implemented.

This is prevalent in other "Command Features" as well.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+
+#### Architecture Overview
```
This seems to be better explained with class diagrams.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+* **Csv**: Handles reading from and writing to the CSV file for data storage persistence.
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
```
A "command flow" seems to suggest a sequence diagram more so than an architecture diagram.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+* handleAddRecord(): handles logic for adding records into the inventory. If successfully add records, data will be appended into the CSV file.
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](docs/umldiagrams/AddCommandClassDiagram.png)
+* **Description**: The class diagram illustrates methods of the `AddCommand` class and the interaction with `Inventory`, `Csv`, and `Ui` respectively.
```
This bullet point is out of place. What is this description for?

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+* handleAddMultipleFields(): processes input fields, validate fields, and update the inventory accordingly.
+* handleAddRecord(): handles logic for adding records into the inventory. If successfully add records, data will be appended into the CSV file.
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](docs/umldiagrams/AddCommandClassDiagram.png)
```
For all diagrams, it would be better to show the diagram first before explaining the details within the diagram.

tzerbin replied:
The navigability seems to suggest that there is a reference of Inventory, Csv, and Ui within the AddCommand class, but this is not the case within the code!

The association label is not indicated with an arrow head.

Please follow proper UML notation for class diagrams.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+#### Sequence Diagram
+*Illustrates how "AddCommand" interacts with "Inventory" and "Csv" classes when adding a record:
+
+![Sequence Diagram](docs/diagrams/AddCommandSequenceDiagram.png)
```
The sequence diagram totally do not match the implementation.
For example, within execute(), handleAddMultipleFields(), handleUpdateFields() and handleAddRecord() is being called based on condition, but the alt-frames are not shown.
There isn't any validateField() method within the class as well.
Only omit return arrows if it does not result in ambiguity.

Are you sure that User is the one who invokes the execute() method?

Please fix all your sequence diagrams within the DG.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,208 @@
+* The system should work across different operating systems (Windows, Linux, macOS).
+* Data must be persistent across sessions by saving and fetching from/to a CSV file.
 
 ## Glossary
```
Glossary is lacking, please double-check on the additional vocabulary that you have introduced, such as "Records".

tzerbin reviewed (commented):
Good attempt on your DG.
However, there is a lot of work that needs to be done, specifically on your diagrams.
Please make sure that the diagrams match your implementation, and that it follows the correct UML notations.

Information such as inheritance was not even included in the class diagrams as well.
Perhaps your team may want to work on some object diagrams too.

Commit:
Merge remote-tracking branch 'Upstream_repo/master'

Commit:
Merge remote-tracking branch 'Upstream_repo/master'

Commit:
Add table of contents for DG

lucas-sc0 committed:
Merge pull request #62 from lucas-sc0/DG-Update

Dg update

tayjerom committed:
Remove add -hu command and update README.md

tayjerom committed:
Merge pull request #63 from tayjerom/v2.0-remove-add-hu

Remove add -hu command and update README.md

tayjerom committed:
Fix Gradle and CI test for Add, View, CommandParser Test

tayjerom committed:
Merge pull request #64 from tayjerom/v2.0-fix-GradleCI-Test

Fix Gradle and CI test for AddCommand, ViewCommand, CommandParser Test

LuxLucky7 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

Commit:
Merge branch 'master' of https://github.com/LuxLucky7/tp

tayjerom committed:
Enhance Coverage Testing for overall code base (Add, Delete, View, Inventra, and Command Parser Test)

tayjerom committed:
Merge pull request #65 from tayjerom/v2.0-Further-ManualTest

Enhance Coverage Testing for overall code base (Add, Delete, View)

tayjerom committed:
Fix delete -h command to update CSV after deleting header

tayjerom committed:
Merge pull request #66 from tayjerom/v2.0-CSVUpdate-delete-h

Fix delete -h command to update CSV after deleting header

Commit:
Implement UpdateCommand

Commit:
Fix UpdateCommand and CommandParser Checkstyle violations

Commit:
UpdateCommand UML diagrams

Commit:
UpdateCommand UserGuide and DeveloperGuide

Commit:
Initial UpdateCommandTest

Commit:
Implement UpdateCommandTest

LuxLucky7 committed:
Merge pull request #67 from LuxLucky7/UpdateCommand

Update Command

Commit:
Merge branch 'master' of https://github.com/LuxLucky7/tp

Commit:
Revise developer guide

Commit:
Merge remote-tracking branch 'Upstream_repo/master' into merge

Commit:
Update UG

Commit:
Add class diagram into DG

Commit:
Fix bugs and crashes

lucas-sc0 committed:
Merge pull request #68 from lucas-sc0/merge

Update DG and fix bugs

Commit:
Update update command input validation

lucas-sc0 committed:
Merge pull request #69 from lucas-sc0/merge

Update update command input validation

tayjerom committed:
Refactor codes identified with bugs upon Manual Test

tayjerom committed:
Merge pull request #70 from tayjerom/v2.0-test-and-bugs-fix-0711

Refactor codes identified with bugs upon Manual Test

tayjerom committed:
Update repository's README.md to reflect Inventra description and Guide.

tayjerom committed:
Update UserGuide.md to reflect consistency of style.

Update User Guide on newly added "UpdateCommand" documentation to fit synchronized documentation style.

Commit:
Update developer guide

lucas-sc0 committed:
Merge pull request #71 from lucas-sc0/merge

Update developer guide

Commit:
Update landing page

Commit:
Update UG

lucas-sc0 committed:
Merge pull request #72 from lucas-sc0/merge

Update landing page

Commit:
Update user guide

lucas-sc0 committed:
Merge branch 'master' into merge

Commit:
Update UG

Commit:
Merge remote-tracking branch 'origin/merge' into merge

lucas-sc0 committed:
Merge pull request #73 from lucas-sc0/merge

Update user guide

Commit:
FIx hyperlinks for UG and DG

lucas-sc0 committed:
Merge pull request #74 from lucas-sc0/master

Update hyperlinks for UG and DG

Commit:
Update UG

lucas-sc0 committed:
Merge pull request #75 from lucas-sc0/master

Update UG

Lambom4n committed:
change error in help commands

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

DucPhong135 committed:
Merge pull request #76 from DucPhong135/master

update help command

Lambom4n committed:
update exception when exit command input more than expected

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

DucPhong135 committed:
Merge pull request #77 from DucPhong135/master

add exception for exit command

Lambom4n committed:
update missing view command in UG

Lambom4n committed:
update on exception message when inputting invalid field type

Lambom4n committed:
Merge remote-tracking branch 'origin/master'

DucPhong135 committed:
Merge pull request #78 from DucPhong135/master

fix minor bugs

tayjerom committed:
Address PE-D issues highlighted #1

tayjerom committed:
Address PE-D issues highlighted #1

tayjerom committed:
Merge pull request #146 from tayjerom/v2.1-fix-PE-D-#1

V2.1 fix pe d #1

tayjerom committed:
Address PE-D issues highlighted #2

danusan-s committed:
Fix issue #90

danusan-s committed:
Remove csv message

danusan-s committed:
Fix UG for delete range

danusan-s committed:
Enforce start is smaller than end in range

danusan-s committed:
Reformat code

danusan-s committed:
Combine text blocks in help command

danusan-s committed:
Fix indentation in help command

danusan-s committed:
Merge pull request #148 from danusan-s/fix-issues-1

Fix issues relating to delete

tayjerom committed:
Address PE-D issues highlighted #3

tayjerom committed:
Merge branch 'master' into v2.1-fix-PE-D-#2

tayjerom committed:
Resolved conflict in CommandParser.java

tayjerom committed:
Merge branch 'v2.1-fix-PE-D-#2' of https://github.com/tayjerom/tp into v2.1-fix-PE-D-#2

tayjerom committed:
Merge pull request #147 from tayjerom/v2.1-fix-PE-D-#2

Address PE-D issues highlighted #2

tayjerom committed:
Merge pull request #149 from tayjerom/v2.1-fix-PE-D-#3

Enhance validateValue method of [AddCommand] and Update UG to remove /n type

tayjerom committed:
Address PE-D issues highlighted #4

tayjerom committed:
Merge branch 'master' into v2.1-fix-PE-D-#4

tayjerom committed:
Merge pull request #150 from tayjerom/v2.1-fix-PE-D-#4

Address PE-D issues highlighted #4

Lambom4n committed:
address issue #81

tayjerom committed:
Address PE-D issues highlighted #5

tayjerom committed:
Merge pull request #151 from tayjerom/v2.1-fix-PE-D-#5

Address PE-D issues highlighted #5

Lambom4n committed:
address issue #120

Lambom4n committed:
address issue #120

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

DucPhong135 committed:
Merge pull request #152 from DucPhong135/master

Fix some issue in PE-D

tayjerom committed:
Update UserGuide.md

tayjerom committed:
Update UserGuide.md

tayjerom committed:
Address issue #153 and included PPP

tayjerom committed:
Merge branch 'master' into v2.1-PPP

tayjerom committed:
Address conflict on InvalidNumberException

tayjerom committed:
Merge pull request #154 from tayjerom/v2.1-PPP

Address issue #153 and included PPP

tayjerom committed:
Address issues raised in PE-D #6

tayjerom committed:
Merge pull request #155 from tayjerom/v2.1-fix-PE-D-#6

Address issues raised in PE-D #6

tayjerom committed:
Address PE-D issues highlighted #7

tayjerom committed:
Merge pull request #156 from tayjerom/v2.1-fix-PE-D-#7

Address PE-D issues highlighted #7

LuxLucky7 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

Commit:
Merge branch 'master' of https://github.com/LuxLucky7/tp

Commit:
Include draft version of luxlucky7.md

Commit:
Fix issue #80

Lambom4n committed:
Update UG and add PPP

Lambom4n committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/UserGuide.md

DucPhong135 committed:
Merge pull request #157 from DucPhong135/master

Update UG and add PPP

Lambom4n committed:
Update view command test

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

DucPhong135 committed:
Merge pull request #158 from DucPhong135/master

Update ViewCommandTest

DucPhong135 committed:
Rename DucPhong135.md to ducphong135.md

DucPhong135 committed:
Rename DucPhong135.md to ducphong135.md

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

danusan-s committed:
Fixed ViewCommand text checkstyle

danusan-s committed:
Added Personal Portfolio

danusan-s committed:
Merge pull request #159 from danusan-s/ppp-danusan

ppp-danusan

Commit:
Resolve issue #145

lucas-sc0 committed:
Merge pull request #160 from lucas-sc0/master

Resolve issues highlighted in PE-D

Commit:
Resolve issue #143

Commit:
Resolve issue #105

Commit:
Add PPP

lucas-sc0 committed:
Merge pull request #161 from lucas-sc0/master

Resolve issue #143

Commit:
Resolve checkstyle error

DucPhong135 committed:
Merge branch 'AY2425S1-CS2113-T11-4:master' into master

Lambom4n committed:
Add exception for input string longer than limit and modify view command to use it
correct duplicate in userguide

Lambom4n committed:
update userguide

DucPhong135 committed:
Merge pull request #163 from DucPhong135/master

Correct UG and modify view -f command to throw exception when the input string exceed 20 characters.

Commit:
Merge remote-tracking branch 'Upstream_repo/master'

Commit:
Fix gradle test

Commit:
FIx checkstyle errors

Commit:
FIx checkstyle errors

Commit:
FIx checkstyle errors

lucas-sc0 committed:
Merge pull request #164 from lucas-sc0/master

Resolve checkstyle error

Commit:
Include final version of luxlucky7.md

LuxLucky7 committed:
Merge pull request #165 from LuxLucky7/docs-team

Docs/team/luxlucky.md

Commit:
Update PPP draft

lucas-sc0 committed:
Merge pull request #166 from lucas-sc0/master

Update PPP draft

Commit:
Update landing page

lucas-sc0 committed:
Merge pull request #167 from lucas-sc0/master

Update landing page

Commit:
Fix landing page link

lucas-sc0 committed:
Merge pull request #168 from lucas-sc0/master

Fix landing page link

Commit:
Fix PPP

lucas-sc0 committed:
Merge pull request #169 from lucas-sc0/master

Fix PPP

Commit:
FIx PPP format

Commit:
Fix PPP

lucas-sc0 committed:
Merge pull request #170 from lucas-sc0/master

FIx PPP format

Commit:
Add Javadoc commments

lucas-sc0 committed:
Merge pull request #171 from lucas-sc0/master

Add Javadoc commments

Commit:
FIx checkstyle errors

lucas-sc0 committed:
Merge pull request #172 from lucas-sc0/master

FIx checkstyle errors

Commit:
Update UG

lucas-sc0 committed:
Merge pull request #173 from lucas-sc0/master

Update UG
